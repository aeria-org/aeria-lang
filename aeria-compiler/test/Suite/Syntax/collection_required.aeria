collection CollectionRequiredTest {
  required {
    prop1
    prop2
    prop3 @cond(prop1 == true)
  }
}
