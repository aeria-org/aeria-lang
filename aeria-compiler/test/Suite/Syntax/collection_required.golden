{
  "kind": "Program",
  "collections": [
    {
      "required": [
        {
          "name": {
            "value": "prop1",
            "kind": "PropertyName"
          },
          "kind": "Required"
        },
        {
          "name": {
            "value": "prop2",
            "kind": "PropertyName"
          },
          "kind": "Required"
        },
        {
          "name": {
            "value": "prop3",
            "kind": "PropertyName"
          },
          "kind": "Required",
          "cond": {
            "kind": "Cond",
            "expr": {
              "right": {
                "kind": "ELiteral",
                "expr": {
                  "value": true,
                  "kind": "LBoolean"
                }
              },
              "left": {
                "kind": "ELiteral",
                "expr": {
                  "value": {
                    "value": "prop1",
                    "kind": "PropertyName"
                  },
                  "kind": "LProperty"
                }
              },
              "kind": "EEq"
            }
          }
        }
      ],
      "name": {
        "value": "collectionrequiredtest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
